src/actions/comment/index.ts

```diff
@@ -377,8 +377,8 @@ const deleteCommentHandler = async (
     }
 
     if (
-      session.user?.role !== ROLES.ADMIN ||
-      existingComment.userId !== userId
+      existingComment.userId !== userId ||
```

Comment:
This is redundant I think,

      (existingComment.userId !== userId && session.user?.role !== ROLES.ADMIN)

should do the work

Reply:
Its not actually , this is where the issue was - 

Actually the problem was , user will always be NOT AN ADMIN , and thats why , it always showed Unauthorized on trying to delete, this AND statement checks for both -> 

1.First , user is not the creater of the comment 
2.Second , user is not an ADMIN as well !